### PR TITLE
[docker] Support host build

### PR DESCRIPTION
**Pull request by WonyoungChoi**
- Install EFL 1.25.1 and dependencies. (including xvfb)
- Fix build-engine.sh for buliding host build.
- Total uncompressed size of the docker will be 2.77GB including the `tizen-tools` image.

```
ghcr.io/flutter-tizen/build-engine   latest    a252ea0a8445   23 minutes ago   2.77GB
ghcr.io/flutter-tizen/tizen-tools    latest    12442f867f85   17 hours ago     1.86GB
```

The build workflows will be:

### build.yml
```yaml
      - name: gclient sync
        run: |
          gclient-prepare-sync.sh --reduce-deps --shallow-sync
          gclient sync -v --no-history --shallow

      - name: build
        run: |
          cache-checksum.sh restore $CACHE_PATH
          build-engine.sh \
            --target-os linux \
            --target-arch ${{ matrix.arch }} \
            --target-triple ${{ matrix.triple }} \
            --runtime-mode ${{ matrix.mode }}          
          cache-checksum.sh save $CACHE_PATH
```

### test.yml
```yaml
      - name: gclient sync
        run: |
          gclient-prepare-sync.sh --reduce-deps --shallow-sync
          gclient sync -v --no-history --shallow

      - name: check format
        run: |
          src/flutter/ci/format.sh

      - name: host build for test
        run: |
          cache-checksum.sh restore $CACHE_PATH
          build-engine.sh --build-target flutter_tizen_unittests
          cache-checksum.sh save $CACHE_PATH

      - name: run unit tests
        run: |
          $CACHE_PATH/flutter_tizen_unittests
```

### CODE REVIEW

File: ci/docker/tizen/Dockerfile [hunk 1]
```diff
@@ -13,3 +25,14 @@ RUN git clone --depth=1 https://chromium.googlesource.com/chromium/tools/depot_t
...
+# Build and install EFL.
+RUN git clone --depth 1 https://git.enlightenment.org/core/efl.git -b -v1.25.1 /tmp/efl && \
+    meson /tmp/efl /tmp/efl/build && \
+    ninja -C /tmp/efl/build && \
+    ninja -C /tmp/efl/build install && \
+    rm -fr /tmp/efl
+ENV LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/lib/x86_64-linux-gnu/
+
+# Start dbus service when running this container.
+ENTRYPOINT /etc/init.d/dbus start && /bin/bash
```
Comment (WonyoungChoi) [inline ci/docker/tizen/Dockerfile:38]: `dbus` service should be started to run `flutter_tizen_unittests`.

File: ci/docker/tizen/Dockerfile [hunk 2]
```diff
@@ -1,8 +1,20 @@
...
+    apt-get install -y libncurses5 libfreetype6-dev && \
+    apt-get install -y build-essential check meson ninja-build && \
+    apt-get install -y libssl-dev libsystemd-dev libjpeg-dev libglib2.0-dev libgstreamer1.0-dev \
+                       liblua5.2-dev libfreetype6-dev libfontconfig1-dev libfribidi-dev libavahi-client-dev \
+                       libharfbuzz-dev libibus-1.0-dev libx11-dev libxext-dev libxrender-dev libgl1-mesa-dev \
+                       libgif-dev libtiff5-dev libpoppler-dev libpoppler-cpp-dev libspectre-dev libraw-dev \
+                       librsvg2-dev libudev-dev libmount-dev libdbus-1-dev libpulse-dev libsndfile1-dev \
+                       libxcursor-dev libxcomposite-dev libxinerama-dev libxrandr-dev libxtst-dev libxss-dev \
+                       libgstreamer-plugins-base1.0-dev doxygen libopenjp2-7-dev libscim-dev libxdamage-dev \
+                       libwebp-dev libunwind-dev libheif-dev libinput-dev libluajit-5.1-dev && \
```
Comment (swift-kim) [inline ci/docker/tizen/Dockerfile:16]: How much image size would be increased by these additional dependencies? Is there no problem?

Comment (WonyoungChoi) [reply ci/docker/tizen/Dockerfile:16]: Compared to `tizen-tools`, the uncompressed size of the added layers is about 1 GB. It takes about 10 more seconds to download the docker image in the build workflows. I think it is acceptable. 
```
ghcr.io/flutter-tizen/build-engine   latest    f1fb55b88802   3 hours ago     2.91GB
ghcr.io/flutter-tizen/tizen-tools    latest    12442f867f85   5 hours ago     1.86GB
```

Comment (swift-kim) [reply ci/docker/tizen/Dockerfile:16]: Oh. That's great.

**Review by swift-kim — APPROVED** [review]

File: ci/docker/tizen/Dockerfile [hunk 3]
```diff
@@ -13,3 +17,33 @@ RUN git clone --depth=1 https://chromium.googlesource.com/chromium/tools/depot_t
...
+RUN apt-get install -y build-essential check meson ninja-build && \
+    apt-get clean
+RUN apt-get install -y libssl-dev libsystemd-dev libglib2.0-dev libudev-dev libmount-dev libdbus-1-dev libunwind-dev && \
+    apt-get clean
+RUN apt-get install -y libjpeg-dev libopenjp2-7-dev libgif-dev libtiff5-dev librsvg2-dev libheif-dev libwebp-dev libraw-dev \
+                       libpoppler-dev libpoppler-cpp-dev libspectre-dev libfreetype6-dev libfontconfig1-dev libharfbuzz-dev \
+                       libpulse-dev libsndfile1-dev libgstreamer1.0-dev libgstreamer-plugins-base1.0-dev \
+                       libibus-1.0-dev libscim-dev libfribidi-dev libinput-dev liblua5.2-dev libluajit-5.1-dev \
+                       libx11-dev libxext-dev libxrender-dev libxcursor-dev libxcomposite-dev libxinerama-dev libxrandr-dev \
+                       libxtst-dev libxss-dev libxdamage-dev libgl1-mesa-dev xvfb && \
```
Comment (bwikbs) [inline ci/docker/tizen/Dockerfile:37]: Wow, now we can run test shell in docker?

**Review by bwikbs — APPROVED** [review]

File: ci/docker/tizen/Dockerfile (hunk 3 above)
Comment (WonyoungChoi) [reply ci/docker/tizen/Dockerfile:37]: Actually, this docker image is for building the `engine` repository and `*-unitests` can be ran after building the engine on this docker. 
#### for example:
```sh
# docker run -it ghcr.io/flutter-tizen/build-engine /bin/bash
mkdir /workspace && cd /workspace
git clone --depth 1 https://github.com/flutter-tizen/engine src/flutter
gclient-prepare-sync.sh --reduce-deps --shallow-sync
gclient sync --no-history
build-engine.sh 
./src/out/host_debug/flutter-tizen-unittests 
```

But if the `test shell` you mentioned refers to the environment for testing the flutter widgets, we will need another docker environment.

**Comment by WonyoungChoi** [comment]
I merge this PR early because of testing #182.

**Pull request merged**